salt/kubernetes-minion/stop.sls

```diff
@@ -0,0 +1,31 @@
+# Stop and disable the Kubernetes minion daemons, ensuring pods have been drained
+# and the node is marked as unschedulable.
+
+# If this, or the state below, fails we should ignore it and proceed anyway as
+# Kubernetes will recover.
+cordon-kubelet:
+  cmd.run:
+    - name: |
+        kubectl --kubeconfig /var/lib/kubelet/kubeconfig cordon {{ grains['id'] }}.{{ pillar['internal_infra_domain'] }}
+    - check_cmd:
+      - /bin/true
+
+drain-kubelet:
+  cmd.run:
+    - name: |
+        kubectl --kubeconfig /var/lib/kubelet/kubeconfig drain {{ grains['id'] }}.{{ pillar['internal_infra_domain'] }} --ignore-daemonsets --grace-period=300 --timeout=340s
```

Comment:
`--ignore-daemonsets` is required, without it, any nodes with a DeamonSet  will just hang on the drain until the timeout hits.

`--force`  would allow for fully removing all pods, however, that's not necessarily what we're going for. Using these flags would result in less pods in the system after the upgrade, i.e. we deleted things which Kubernetes can't resurrect itself.

We're instead migrating anything we can away from this node, but manual pods can't be migrated. For those pods, which are hopefully really really rare, we just want to let it fail/timeout on these, continue anyway, and kubernetes will resurrect them on the same node when it starts back up.